executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -32,6 +133,23 @@ data State
 
 -- | Ideation signals.
 data Signal
-  = Submit SIP
+  = Submit SIPCommit SIP
```

Comment:
Can't we just submit a `SIP` and calculate the `Commit` from it (We have the `SIP`, `pk`, and `salt`).

Reply:
This is true. I thought that it resembles better the actual signal. In the real case, only the committed SIP is submitted. I can change it.

Reply:
Maybe we can leave it for now, since we need a way to determine to which SIP a commit belongs, so your approach might be right here.

Reply:
At a second thought. maybe we should have a separate SIP Commit data structure. This includes - apart from the commit field - also the signature of the author and at a future version, we could also add signature verification in the Reveal rules.